src/recon_test/test_consistency_with_root.cxx

```diff
@@ -0,0 +1,547 @@
+   * read only once.
+   *
+   * Passes ProjMatrixElemsForOneBin (LOR) to test_nonTOF_LOR_closest_approach() and if fails,
+   * add 1 to `failed_events` (LOR's closest voxel was not within nonTOF_distance_threshold).
+   * Check if the number of `failed_events` is greater than half the number of tested events to pass the test.
+   */
+  void process_list_data();
+
+  void loop_through_list_events(shared_ptr<CListModeData> lm_data_sptr, const shared_ptr<ProjMatrixByBin>& proj_matrix_sptr,
+                                const CartesianCoordinate3D<float> original_coords);
+  /*! Given a ProjMatrixElemsForOneBin (probabilities), test if the closest voxel in the LOR to the original_coords
+   * is within nonTOF_distance_threshold distance. If it is, pass with true, otherwise fales.
+   * @param probabilities ProjMatrixElemsForOneBin object of a list mode event
+   * @param test_discretised_density_sptr Density containing a point source.
+   * @param original_coords Precomputed coordinates of the point source
+   * @param tolerance Precomputed voxel sizes - used in setting nonTOF_distance_threshold
+   * @return True if test passes, false if failed.
+   */
+  bool test_nonTOF_LOR_closest_approach(const ProjMatrixElemsForOneBin& probabilities);
+  void post_processing_nonTOF();
+
+
+  //! Selects and stores the highest probability elements of ProjMatrixElemsForOneBin.
+  bool test_TOF_max_lor_voxel(const ProjMatrixElemsForOneBin& probabilities);
+
+  //! Checks if original and calculated coordinates are close enough and saves data to file.
+  void post_processing_TOF();
+
+  //! Modified version of check_if_equal for this test
+  bool check_if_almost_equal(const float a, const float b, const std::string str, const float tolerance);
+
+  // Auxilary methods
+  std::string get_root_header_filename() { return "pretest_output/root_header_test" + std::to_string(test_index) + ".hroot"; }
+  std::string get_generate_image_par_filename() { return "SourceFiles/generate_image" + std::to_string(test_index) + ".par"; }
+
+
+  // Class VARIABLES
+
+  int test_index;
+  int num_tests;
+  std::vector<bool> test_results_nonTOF;
+  std::vector<bool> test_results_TOF;
+
+  shared_ptr<DiscretisedDensity<3, float> > discretised_density_sptr;
+  CartesianCoordinate3D<float> original_coords; // Stored in class because of dynamic_cast
+  CartesianCoordinate3D<float> grid_spacing;    // Stored in class because of dynamic_cast
+
+  int num_events_tested = 0;
+  const float failure_tolerance_nonTOF = 0.05;
+  const float failure_tolerance_TOF = 0.05;
+
+  /// NON TOF VARIABLES
+  //! A vector to store the coordinates of the closest approach voxels.
+  std::vector<CartesianCoordinate3D<float>> nonTOF_closest_voxels_list;
+  int num_failed_nonTOF_lor_events = 0;
+  double nonTOF_distance_threshold;
+
+  /// TOF VARIABLES
+  //! A copy of the scanner timing resolution in mm
+  float gauss_sigma_in_mm;
+  std::vector<WeightedCoordinate> max_coords_for_each_event;
```

Comment:
`max_coords_for_each_event` is a misleading variable name. Consider `LOR_peak_value_coords`? Peak is more descriptive and LOR is clearer than event.

Reply:
Add `TOF` to indicate TOF test variable -> `TOF_LOR_peak_value_coords`